Simplified/Accounts/AgeCheck/NYPLAgeCheck.swift

```diff
@@ -0,0 +1,120 @@
+import Foundation
+
+protocol NYPLAgeCheckValidationDelegate: class {
+  var minYear : Int { get }
+  var currentYear : Int { get }
+  var birthYearList : [Int] { get }
+  var ageCheckCompleted : Bool { get set }
+  
+  func isValid(birthYear: Int) -> Bool
+  
+  func didCompleteAgeCheck(_ birthYear: Int)
+  func didFailAgeCheck()
+}
+
+@objc protocol NYPLAgeCheckVerifying {
+  func verifyCurrentAccountAgeRequirement(userAccountProvider: NYPLUserAccountProvider,
+                                          currentLibraryAccountProvider: NYPLCurrentLibraryAccountProvider,
+                                          completion: ((Bool) -> ())?) -> Void
+}
+
+@objc protocol NYPLAgeCheckChoiceStorage {
+  var userPresentedAgeCheck: Bool { get set }
+}
+
+@objcMembers final class NYPLAgeCheck : NSObject, NYPLAgeCheckValidationDelegate, NYPLAgeCheckVerifying {
+  
+  // Members
+  private let serialQueue = DispatchQueue(label: "\(Bundle.main.bundleIdentifier!).ageCheck")
+  private var handlerList = [((Bool) -> ())]()
+  private var isPresenting = false
+  private let ageCheckChoiceStorage: NYPLAgeCheckChoiceStorage
+  var ageCheckCompleted: Bool = false
+  
+  let minYear: Int
+  let currentYear: Int
+  let birthYearList: [Int]
+
+  init(ageCheckChoiceStorage: NYPLAgeCheckChoiceStorage) {
+    self.ageCheckChoiceStorage = ageCheckChoiceStorage
+    minYear = 1900
+    currentYear = Calendar.current.component(.year, from: Date())
+    birthYearList = Array(minYear...currentYear)
+    
+    super.init()
+  }
+  
+  func verifyCurrentAccountAgeRequirement(userAccountProvider: NYPLUserAccountProvider,
+                                          currentLibraryAccountProvider: NYPLCurrentLibraryAccountProvider,
+                                          completion: ((Bool) -> ())?) {
+    serialQueue.async {
```

Comment:
not sure how I missed this but there's a big retain cycle here:

self->serialQueue->closure -> self